spec/statsd/connection_cfg_spec.rb

```diff
@@ -0,0 +1,218 @@
+        expect(subject.socket_path).to eq '/some/socket'
+      end
+    end
+
+    context 'with no args and DD_AGENT_HOST set' do
+      let(:dd_agent_host) { 'some-host' }
+
+      it 'creates a UDP connection' do
+        expect(subject.transport_type).to eq :udp
+      end
+
+      it 'sets host to DD_AGENT_HOST' do
+        expect(subject.host).to eq 'some-host'
+      end
+
+      it 'sets port to 8125 (default)' do
+        expect(subject.port).to eq 8125
+      end
+
+      it 'sets socket_path to nil' do
+        expect(subject.socket_path).to eq nil
+      end
+
+      context 'and DD_DOGSTATSD_PORT set' do
+        let(:dd_dogstatsd_port) { '1234' }
+
+        it 'creates a UDP connection' do
+          expect(subject.transport_type).to eq :udp
+        end
+
+        it 'sets host to DD_AGENT_HOST' do
+          expect(subject.host).to eq 'some-host'
+        end
+
+        it 'sets port to DD_DOGSTATSD_PORT' do
+          expect(subject.port).to eq 1234
+        end
+
+        it 'sets socket_path to nil' do
+          expect(subject.socket_path).to eq nil
+        end
+      end
+    end
+
+    context 'with no args and DD_DOGSTATSD_SOCKET set' do
+      let(:dd_dogstatsd_socket) { '/some/socket' }
+
+      it 'creates a UDS connection' do
+        expect(subject.transport_type).to eq :uds
+      end
+
+      it 'sets host to nil' do
+        expect(subject.host).to eq nil
+      end
+
+      it 'sets port to nil' do
+        expect(subject.port).to eq nil
+      end
+
+      it 'sets socket_path to DD_DOGSTATSD_SOCKET' do
+        expect(subject.socket_path).to eq '/some/socket'
+      end
+    end
+
+    context 'with both DD_AGENT_HOST and DD_DOGSTATSD_SOCKET set' do
+      let(:dd_agent_host) { 'some-host' }
+      let(:dd_dogstatsd_socket) { '/some/socket' }
+
+      it 'raises an exception' do
+        expect do
+          subject.new(host: host, port: port, socket_path: socket_path)
+        end.to raise_error(ArgumentError, 'Do not set both host/port and socket_path')
+      end
+    end
+
+    context 'with no args and no env vars set' do
+      it 'creates a UDP connection' do
+        expect(subject.transport_type).to eq :udp
+      end
+
+      it 'sets host to 127.0.0.1 (default)' do
+        expect(subject.host).to eq '127.0.0.1'
+      end
+
+      it 'sets port to 8125 (default)' do
+        expect(subject.port).to eq 8125
+      end
+
+      it 'sets socket_path to nil' do
+        expect(subject.socket_path).to eq nil
+      end
+    end
+  end
+
+  describe '#make_connection' do
+    context 'for a UDP connection' do
+      before do
+        allow(Datadog::Statsd::UDPConnection)
+          .to receive(:new)
+          .with(host, port, param: 'param')
+          .and_return(udp_connection)
+      end
+
+      let(:host) { 'my-agent' }
+      let(:port) { 1234 }
+      let(:udp_connection) do
+        instance_double(Datadog::Statsd::UDPConnection,
+          host: host,
+          port: port
```

Comment:
Minor: Since the host/port on the instance double don't seem to be used, they can be removed.